pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -622,3 +627,7 @@ func (r *Reconciler) matchControllerRequirement(app *v1beta1.Application) bool {
 	}
 	return true
 }
+
+func (r *Reconciler) matchAppSharding(object client.Object) bool {
+	return object.GetLabels()[oam.LabelAppSharding] == pflag.Lookup("leader-election-name").Value.String()
```

Comment:
don't look up every time, the performance could be better

Reply:
> don't look up every time, the performance could be better

ok, i will fix it in recently